lib/handlers/post.js

```diff
@@ -80,20 +81,19 @@ function handler (req, res, next) {
 
   function one () {
     debug('Receving one file')
-    var linkHeader = header.parseMetadataFromHeader(req.get('Link'))
-    var slug = req.get('Slug')
-    ldp.post(
-      req.hostname,
-      containerPath,
-      slug,
-      req,
-      linkHeader.isBasicContainer,
+    const { slug, link, 'content-type': contentType } = req.headers
+    const links = header.parseMetadataFromHeader(link)
+    const mimeType = contentType ? contentType.replace(/\s*;.*/, '') : ''
+    const extension = mimeType in extensions ? `.${extensions[mimeType][0]}` : ''
+
+    ldp.post(req.hostname, containerPath, req,
+      { slug, extension, container: links.isBasicContainer },
       function (err, resourcePath) {
         if (err) {
           return next(err)
         }
         debug('File stored in ' + resourcePath)
-        header.addLinks(res, linkHeader)
+        header.addLinks(res, links)
```

Comment:
Why do we send back the link headers in the response?

Reply:
I have no idea; just preserving what was there…